feat: improve report screen

JonathanPilemand opened the pull request:
## Description
This PR improves the report screen by changing the layout and the functionnality. The report page feed has changed, as well as the individual report items. Some actions have been removed, and some refactoring has been made to account for these new changes. 

### New `ReportSceen` and `ReportScreenViewModel`
The report screen is different from the home screen in that the report items don't take up all the width, but they have similar header and image structure. To differentiate the reports from the posts, we now also have the status information and location shown above and below the image respectively, as well as the expandable description. That means the report action buttons are now removed from the page, as it makes more sense to have them in the report detail page, where they're also better implemented. This means the underlying methods in the view model have also been removed. Moreover, the ui state exposed by the viewmodel no longer includes the `assigneeUsername`, but just an `assigned` filed to indicate the report status. This already gives the user an idea of what report actions will available on the report details screen. 

As introduced in the home screen, the report screen now also has a slider to swipe between the report image and the report mini map. The difference is that since reports always have a location, the slider (and mini map) will always be shown.

Lastly, the submit report button has been changed to no longer be a big button centered at the bottom, but a button that appears when clicking on the more actions button in the bottom right corner of the page. The add icon rotates to a cross icon and reveals the submit report buttons by sliding it up using the new `ReportScreenButtons` composable. 


### Refactoring
`ReportScreenTest`, `ReportScreenViewModelTest` and `NavigationTestUtils` have been refactored to include these changes. The `ExpandableText` composable now takes default arguments for the body and toggle tags. 
Apart from the buttons logic, some of the error handling in `ReportScreenViewModel` has also been changed. We now only use the `defaultUser` for the intial ui state like in the home screen, and have removed internal error handling in the update method.

---
<p align="center"><strong>Here are some examples of the new report screen</strong></p>

<p align="center">
<img src="https://github.com/user-attachments/assets/1ded96e0-9b76-4640-92a8-52664fca795c" width="175" alt="Screenshot preview 1">
<img src="https://github.com/user-attachments/assets/90136c75-c615-4f4a-bfac-a970766b2924" width="175" alt="Screenshot preview 2">
<img src="https://github.com/user-attachments/assets/57d4f7a6-a199-4e6d-88f9-9da0c63a19ca" width="175" alt="Screenshot preview 3">
<img src="https://github.com/user-attachments/assets/08e3ee04-e443-491e-844a-52a20ff40476" width="175" alt="Screenshot preview 4">
</p>


---

## Related issues
Closes #315 

## Trade-offs or known limitations 
- The images are fixed sized and therefore sometimes zoomed, this will probably change in a subsequent PR, like in the home screen

## How to test
Run `ReportScreenTest`, `ReportScreenViewModelTest` and `NavigationTest`. Check that other tests still work

Rania5724 commented on app/src/main/java/com/android/wildex/ui/report/ReportScreenViewModel.kt:
this seems a little useless. Since you set `isError `to false here

```
_uiState.value =
          _uiState.value.copy(
              reports = reportUIStates,
              isLoading = false,
              isRefreshing = false,
              isError = false,
          )
```

this error message will never be displayed so its best to remove it and just silently display an empty list if the repo call fails.

JonathanPilemand replied:
We'll do the same as for the user

Rania5724 reviewed (changes requested):
I love the new submit report button design, it looks much better. Also good job for the line coverage and having no sonar issues. However I admit I am not a big fan of the current UI design:

1.The report images seem to have a fixed size so some images appear cropped, some have dark edges so I suggest you implement the same logic you are planning to do for home with max and min height.

<img width="364" height="413" alt="Image" src="https://github.com/user-attachments/assets/ecd2b9ca-6ae9-4f0c-b606-dcc000aac96f" />

2. The close/assigned buttons' design looks.....interesting, and their placement with the location just makes the middle look very empty.
3. I would prefer if we changed a little bit from the "instagram" design that we had in the home screen, as it doesnt really fit the whole report vibes. Honestly, I would prefer the old design to this, now that we removed the buttons.
4. For the UI design, I would maybe propose doing something like to change a little (and I will also comment the code I implemented if you like the idea, otherwise let's discuss this)

<p align="center">
<img src="https://github.com/user-attachments/assets/ef66b3cd-d41c-40a7-a7e8-1afef42ffddd" width="175" alt="Screenshot preview 1">
<img src="https://github.com/user-attachments/assets/cd9e0bd5-3f62-4276-a906-a5414787029c" width="175" alt="Screenshot preview 2">
</p>

5. I also have some code logic comments see below

JonathanPilemand commented:
Thank you for the feedback, I've applied the necessary changes and updated the PR description accordingly. That includes the logic and most of the UI suggestions.
However, for the images:
- The dark borders around the dog is simply the picture itself (it's not a good test image to use, I've changed it)
- It is true that they're fixed size and zoomed, as done currently in the home page. The plan is to fix that, but this is still under development for the home page, so it will be implemented for the reports once I've done that. That means it will be in another PR

This one should therefore be ready now
